apps/dashboard/src/@3rdweb-sdk/react/components/connect-wallet/index.tsx

```diff
@@ -1,85 +1,182 @@
+        const res = await fetch(`${THIRDWEB_API_HOST}/v1/auth/payload`, {
+          method: "POST",
+          body: JSON.stringify({
+            address: params.address,
+            chainId: params.chainId.toString(),
+          }),
+          headers: {
+            "Content-Type": "application/json",
+          },
+        });
+        if (!res.ok) {
+          throw new Error("Failed to fetch login payload");
+        }
+        return (await res.json()).payload;
+      },
+      doLogin: async (payload) => {
+        const res = await fetch(`${THIRDWEB_API_HOST}/v1/auth/login`, {
+          method: "POST",
+          body: JSON.stringify({ payload }),
+          credentials: "include",
+          headers: {
+            "Content-Type": "application/json",
+          },
+        });
+        if (!res.ok) {
+          throw new Error("Failed to login");
+        }
+        const json = await res.json();
+
+        if (json.token) {
+          // biome-ignore lint/suspicious/noExplicitAny: FIXME
+          (window as any)[GLOBAL_AUTH_TOKEN_KEY] = json.token;
+          queryClient.invalidateQueries({
+            queryKey: ["logged_in_user"],
+          });
+        }
+        return json;
+      },
+      doLogout: async () => {
+        // reset the token ASAP
+        // biome-ignore lint/suspicious/noExplicitAny: FIXME
+        (window as any)[GLOBAL_AUTH_TOKEN_KEY] = undefined;
+        const res = await fetch(`${THIRDWEB_API_HOST}/v1/auth/logout`, {
+          method: "POST",
+          credentials: "include",
+          headers: {
+            "Content-Type": "application/json",
+          },
+        });
+        if (!res.ok) {
+          throw new Error("Failed to logout");
+        }
+        return res.json();
+      },
+      isLoggedIn: async (address) => {
+        const res = await fetch(`${THIRDWEB_API_HOST}/v1/auth/user`, {
+          method: "GET",
+          credentials: "include",
+          headers: {
+            "Content-Type": "application/json",
+          },
+        });
+        if (!res.ok) {
+          return false;
+        }
+
+        const { jwt } = await fetchAuthToken(address);
+        if (jwt) {
+          // biome-ignore lint/suspicious/noExplicitAny: FIXME
+          (window as any)[GLOBAL_AUTH_TOKEN_KEY] = jwt;
+          queryClient.invalidateQueries({
+            queryKey: ["logged_in_user"],
+          });
+          return true;
+        }
+
+        return false;
+      },
+    } satisfies SiweAuthOptions;
+  }, [noAuth, queryClient]);
+
   return (
-    <ConnectWallet
-      auth={auth}
+    <ConnectButton
+      auth={authConfig}
       theme={t}
-      welcomeScreen={() => {
-        return <ConnectWalletWelcomeScreen theme={t} />;
+      client={thirdwebClient}
+      connectModal={{
+        privacyPolicyUrl: "/privacy",
+        termsOfServiceUrl: "/tos",
+        showThirdwebBranding: false,
+        welcomeScreen: () => <ConnectWalletWelcomeScreen theme={t} />,
```

Comment:
btw not right now, but i think short term we might want to get rid of the welcome screen to just promote the inApp wallet login more